engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -112,10 +114,25 @@ class EngineJobExecutionActor(jobKey: BackendJobDescriptorKey,
       writeToMetadata(Map(callCachingReadResultMetadataKey -> "Cache Miss"))
       log.info(s"Cache miss for job ${jobDescriptor.key.call.fullyQualifiedName}, index ${jobDescriptor.key.index}")
       runJob(jobDescriptor, bjeaProps)
-    case Event(CacheHit(cacheResultId), JobDescriptorData(jobDescriptor, _)) =>
+    case Event(CacheHit(cacheResultId), JobDescriptorData(jobDescriptor, bjeaProps)) =>
       writeToMetadata(Map(callCachingReadResultMetadataKey -> s"Cache Hit (from result ID $cacheResultId)"))
       log.info(s"Cache hit for job ${jobDescriptor.key.call.fullyQualifiedName}, index ${jobDescriptor.key.index}! Copying cache result $cacheResultId")
-      lookupCachedResult(jobDescriptor, cacheResultId)
+      lookupCacheResult(jobDescriptor, bjeaProps, jobKey.call.task.outputs, cacheResultId)
+  }
+
+  // When PreparingCachedOutputs, the FSM should have EJEAJobDescriptorData
+  when(PreparingCachedOutputs) {
+    case Event(CachedOutputLookupSucceeded(cachedJobOutputs), JobDescriptorData(jobDescriptor, _)) =>
+      //I can remove some of this logging once this PR is reviewed--mostly for my own debugging
```

Comment:
Meh you can get rid of it even now, if the reviewers feel like the things being logged are not excessive. I was wondering if some of the EJEA logging should be debug mode specific